ManalHamdi/assignment4/Assignment4.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <utility>
+#include <queue>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+bool isValidNeighbour(const std::vector<std::vector<bool> > &map_tiles, int row, int column, int number_rows, int number_columns) {
+	return (map_tiles[row][column] && row >= 0 && row < number_rows && column >= 0 && column < number_columns);
+}
+std::vector<std::pair<int, int> > getNeighboursIndices(const std::vector<std::vector<bool> > &map_tiles, int row, int column, int number_rows, int number_columns) {
```

Comment:
When you have a reaaally long line of code, consider splitting it over multiple lines to make it easier to read for anyone without a 2 meter wide monitor :) When writing code at Google, we have very strict line length limits, you literally can't submit your code the line is too long :)

This is actually tricky as you want to make the code look nice even if you have to split it. For this line you can do something like this:

```
std::vector<std::pair<int, int> > getNeighboursIndices(
                                      const std::vector<std::vector<bool> > &map_tiles,
                                      int row, int column, int number_rows,
                                      int number_columns) {
```

Reply:
The same for isValidNeighbour, BFS and one of the tests :)